bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/CaddxBaseThingHandler.java

```diff
@@ -0,0 +1,294 @@
+     *
+     * @param caddxThingType The Thing type
+     */
+    private void getConfiguration(CaddxThingType caddxThingType) {
+        switch (caddxThingType) {
+            case PARTITION:
+                CaddxPartitionConfiguration partitionConfiguration = getConfigAs(CaddxPartitionConfiguration.class);
+                setPartitionNumber(partitionConfiguration.getPartitionNumber().intValue());
+                break;
+            case ZONE:
+                CaddxZoneConfiguration zoneConfiguration = getConfigAs(CaddxZoneConfiguration.class);
+                setZoneNumber(zoneConfiguration.getZoneNumber().intValue());
+                break;
+            case KEYPAD:
+                CaddxKeypadConfiguration keypadConfiguration = getConfigAs(CaddxKeypadConfiguration.class);
+                setKeypadAddress(keypadConfiguration.getKeypadAddress().intValue());
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Get the Thing type.
+     *
+     * @return caddxThingType
+     */
+    public CaddxThingType getCaddxThingType() {
+        return caddxThingType;
+    }
+
+    /**
+     * Get Partition Number.
+     *
+     * @return partitionNumber
+     */
+    public int getPartitionNumber() {
+        return partitionNumber;
+    }
+
+    /**
+     * Set Partition Number.
+     *
+     * @param partitionNumber
+     */
+    public void setPartitionNumber(int partitionNumber) {
+        this.partitionNumber = partitionNumber;
+    }
+
+    /**
+     * Get Zone Number.
+     *
+     * @return zoneNumber
+     */
+    public int getZoneNumber() {
+        return zoneNumber;
+    }
+
+    /**
+     * Set Zone Number.
+     *
+     * @param zoneNumber
+     */
+    public void setZoneNumber(int zoneNumber) {
+        this.zoneNumber = zoneNumber;
+    }
+
+    /**
+     * Get Keypad Address.
+     *
+     * @return keypadAddress
+     */
+    public int getKeypadAddress() {
+        return keypadAddress;
+    }
+
+    /**
+     * Set Keypad Address.
+     *
+     * @param keypadAddress
+     */
+    public void setKeypadAddress(int keypadAddress) {
+        this.keypadAddress = keypadAddress;
+    }
+
+    /**
+     * Get Channel by ChannelUID.
+     *
+     * @param channelUID
+     */
+    public @Nullable Channel getChannel(ChannelUID channelUID) {
+        Channel channel = null;
+
+        List<Channel> channels = getThing().getChannels();
+
+        for (Channel ch : channels) {
+            if (channelUID == ch.getUID()) {
+                channel = ch;
+                break;
+            }
+        }
+
+        return channel;
+    }
+
+    /**
+     * Get Thing Handler refresh status.
+     *
+     * @return thingRefresh
+     */
+    public boolean isThingHandlerInitialized() {
+        return thingHandlerInitialized;
+    }
+
+    /**
+     * Set Thing Handler status.
+     *
+     * @param deviceInitialized
+     */
+    public void setThingHandlerInitialized(boolean deviceInitialized) {
```

Comment:
Why isn't this called in the initialize() method?

Reply:
It is not used actually. Copy leftover from the old binding.
I deleted the field and the relevant code.